### PR TITLE
[FIX] WIndows binary

### PR DESCRIPTION
**Pull request by dupasj**
I encountered issues when running the `node-jq` binary from the package. I discovered that if you don't append `.exe` to the `bin` package file path, the binary isn't added/symlinked in the `node_module/.bin/` repository, resulting in the binary not being resolved.

In response to this issue, I've added the `renamePackageBin` function, which forces to assign the full filename as the binary file in the `package.json`, resolving the problem.

I've executed this function in all OS contexts, which should work, but we can consider guarding it for Windows only (when the filename ends by `.exe`, for example).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,86 +1,86 @@
...
-    "validate-commit-msg": "^2.14.0"
-  },
-  "standard": {
-    "global": [
-      "describe",
-      "it"
-    ]
-  }
-}
+	"name": "node-jq",
```
Comment (davesnx) [inline package.json:2]: Could you revert this file changes?

File: package.json [hunk 2]
```diff
@@ -1,86 +1,86 @@
...
+	"version": "0.0.0-semantic-release",
+	"description": "Run jq in node",
+	"main": "lib/jq.js",
+	"repository": {
+		"type": "git",
+		"url": "https://github.com/sanack/node-jq"
+	},
+	"bin": {
+		"node-jq": "bin/jq.exe"
```
Comment (davesnx) [inline package.json:11]: That's rather annoying to add here, since it will break linux-based OS. Please remove this diff

File: scripts/install-binary.js [hunk 3]
```diff
@@ -40,9 +40,20 @@ const JQ_NAME_MAP = {
...
 const JQ_NAME =
-  platform in JQ_NAME_MAP ? JQ_NAME_MAP[platform] : JQ_NAME_MAP.def
+    platform in JQ_NAME_MAP ? JQ_NAME_MAP[platform] : JQ_NAME_MAP.def
 
+const PACKAGE_FOLDER = path.join(__dirname, '..')
+const PACKAGE_FILE = path.join(PACKAGE_FOLDER, 'package.json')
 const OUTPUT_DIR = path.join(__dirname, '..', 'bin')
+const OUTPUT_FILE = path.join(OUTPUT_DIR, JQ_NAME)
+
+const renamePackageBin = () => {
```
Comment (davesnx) [inline scripts/install-binary.js:50]: Can you rename this function to "makeBinaryWorkInWindows"? and add comments on all things happening here? Replace `\` with `/` and also implement the append of ".exe"?

**Review by davesnx — COMMENTED** [review]
Can we make sure this works fine in Windows on CI?

**Comment by dupasj** [comment]
> Can we make sure this works fine in Windows on CI?

I do not have access to your CI or I cannot see the CI, can you please me prompt me the error if you have ?

File: scripts/install-binary.js (hunk 3 above)
Comment (dupasj) [reply scripts/install-binary.js]: I resolved:

**Comment by dupasj** [comment]
Hi @davesnx !

After couple of tests, I figured out that if you install again the dependency the binary file is welly symbolinked. 

It's look like the binary symbolink aare created before the preinstall hooks (so if the file is not already downloaded, the symbolink is ignored.) So I force to run "npm link ." in the package folder after the file download (take a look to this [commit](https://github.com/sanack/node-jq/pull/671/commits/30cab69668d5b5aaa438fc623a9ba3597d62e89d))

Note that I also change the test unit: I had to replace "null" by "undefined" at some points.

**Comment by davesnx** [comment]
The preinstall hook should fix this issue since https://github.com/sanack/node-jq/pull/670. If there's an issue with that, please open an issue.
